ivadomed/loader/bids_dataframe.py

```diff
@@ -86,25 +87,25 @@ def create_bids_dataframe(self):
             ext_ct = ('.nii.gz', '.nii')
             suffix_ct = ('ct', 'CT')
             force_index = []
-            for root, dirs, files in os.walk(path_data):
-                for file in files:
+            for path_object in path_data.glob('**/*'):
+                if path_object.is_file():
                     # Microscopy
-                    if file == "samples.tsv" or file == "samples.json":
-                        force_index.append(file)
-                    if (file.endswith(ext_microscopy) and os.path.basename(root) == "microscopy" and
-                            (root.replace(path_data, '').startswith("sub"))):
-                        force_index.append(os.path.join(root.replace(path_data, '')))
+                    if path_object.name == "samples.tsv" or path_object.name == "samples.json":
+                        force_index.append(path_object.name)
+                    if (path_object.name.endswith(ext_microscopy) and path_object.parent.name == "microscopy" and
+                            (path_object.parts[len(path_data.parts)]).startswith('sub')):
```

Comment:
Very cool. Didn't know pathlib had `parts` for tuples. 

Line 96 is a bit too complex for one liner. Is there a way to make it more clear? I suggest expand it out a bit. 
The goal here is to get to the subject path. Which should be the last  part of path_data. I am not sure what edge case the code originally was anticipating. Maybe @mariehbourget has more insight on the explicit 'sub' prefix check here.